sequencing-server/src/lib/filters/utilities.ts

```diff
@@ -0,0 +1,272 @@
+            if (typeof filterValue === 'string' && typeof itemValue === 'string') {
+                if (filterValue === '') {
+                    return true;
+                }
+                return itemValue.indexOf(filterValue) < 0;
+            } else if (Array.isArray(filterValue)) {
+                return !(Array.isArray(itemValue) ? itemValue : [itemValue]).find(
+                    item => (filterValue as (typeof itemValue)[]).indexOf(item) > -1,
+                );
+            }
+            return false;
+        case 'is_greater_than':
+            return itemValue > filterValue;
+        case 'is_less_than':
+            return itemValue < filterValue;
+        case 'is_within':
+            if (
+                Array.isArray(filterValue) &&
+                filterValue.length === 2 &&
+                typeof filterValue[0] === 'number' &&
+                typeof filterValue[1] === 'number'
+            ) {
+                // TODO should upper bound be inclusive or exclusive?
+                return itemValue >= filterValue[0] && itemValue <= filterValue[1];
+            }
+            return false;
+        case 'is_not_within':
+            if (
+                Array.isArray(filterValue) &&
+                filterValue.length === 2 &&
+                typeof filterValue[0] === 'number' &&
+                typeof filterValue[1] === 'number'
+            ) {
+                // TODO should upper bound be inclusive or exclusive?
+                return itemValue < filterValue[0] || itemValue > filterValue[1];
+            }
+            return false;
+        default:
+            return false;
+    }
+}
+
+export function getMatchingTypesForActivityLayerFilter(filter: ActivityLayerFilter | undefined, types: ActivityType[]) {
+    if (
+        !filter ||
+        (!filter.dynamic_type_filters?.length &&
+            !filter.other_filters?.length &&
+            !filter.static_types?.length &&
+            (!filter.type_subfilters || !Object.keys(filter.type_subfilters).length))
+    ) {
+        return types;
+    }
+
+    const staticTypeMap: Record<string, boolean> = (filter.static_types || []).reduce(
+        (acc: Record<string, boolean>, cur: string) => {
+            acc[cur] = true;
+            return acc;
+        },
+        {},
+    );
+
+    const anyTypeFiltersSpecified = !!(filter.static_types?.length || filter.dynamic_type_filters?.length);
+
+    return types.filter(type => {
+        let included = !anyTypeFiltersSpecified;
+
+        // Check to see if type is included in static list
+        if (filter.static_types?.length) {
+            included = !!staticTypeMap[type.name];
+        }
+
+        // Check if necessary to see if type is included in dynamic list
+        if ((!filter.static_types?.length || !included) && filter.dynamic_type_filters?.length) {
+            included = typeMatchesDynamicFilters(type, filter.dynamic_type_filters);
+        }
+        return included;
```

Comment:
After a bit of boolean algebra, this is equivalent to the following:
```suggestion
    const includedInStaticTypes = !filter.static_types?.length || !!staticTypeMap[type.name];
    const includedInDynamicTypes = !filter.dynamic_type_filters?.length || typeMatchesDynamicFilters(type, filter.dynamic_type_filters);
    return includedInStaticTypes && includedInDynamicTypes;
```

Reply:
De Morgan would be proud